common/lib/dependabot/dependency_group.rb

```diff
@@ -1,25 +1,53 @@
 # frozen_string_literal: true
 
+require "dependabot/experiments"
+require "dependabot/config/ignore_condition"
+require "dependabot/logger"
+
 require "wildcard_matcher"
 require "yaml"
 
 module Dependabot
   class DependencyGroup
+    ANY_DEPENDENCY_NAME = "*"
+    SECURITY_UPDATES_ONLY = false
+
+    SEMVER_MAJOR = "major"
+    SEMVER_MINOR = "minor"
+    SEMVER_PATCH = "patch"
+    DEFAULT_SEMVER_LEVEL = SEMVER_MINOR
+
+    class NullIgnoreCondition
+      def ignored_versions(_dependency, _security_updates_only)
+        []
+      end
+    end
+
     attr_reader :name, :rules, :dependencies
 
     def initialize(name:, rules:)
       @name = name
       @rules = rules
       @dependencies = []
+      @ignore_condition = generate_ignore_condition!
```

Comment:
Pedantic, but maybe we can make the naming explicitly say what these ignore conditions are for?